compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1002,14 +1008,6 @@ fn strip_symbols_in_osx<'a>(sess: &'a Session, out_filename: &Path, option: &str
     }
 }
 
-fn osx_strip_opt<'a>(strip: Strip) -> Option<&'a str> {
-    match strip {
-        Strip::Debuginfo => Some("-S"),
-        Strip::Symbols => Some("-x"),
```

Comment:
`-x` strips local symbols, but keeps global symbols. I think global symbols are necessary for linking to dylibs and thus must not be stripped.

Reply:
According to the manpage strip without any options is smart. If the executable uses dylibs strip without any options and the executable does use dylibs it is equivalent to `strip -u -r` _"which saves all undefined symbols (usually defined in the dynamic shared libraries) and all global  symbols  defined  in  the  executable  referenced  by  the  dynamic libraries  (as  marked  by  the  static link editor when the executable was built). This is the maximum level of striping for an executable that will still  allow  the program to run correctly with its libraries."_